cluster-topology.md

```diff
@@ -0,0 +1,273 @@
+
+**Table of Contents**
+
+- [Scale out control plane nodes](#scale-out-control-plane-nodes)
+  - [Scale from 1 to 3 control plane nodes](#scale-from-1-to-3-control-plane-nodes)
+- [More scale operations](#more-scale-operations)
+- [Add a new pool of worker nodes](#add-a-new-pool-of-worker-nodes)
+- [Remove a worker node](#remove-a-worker-node)
+- Next: [Changing Cluster Topology](#changing-cluster-topology)
+
+## What is Cluster Topology?
+
+We define a cluster topology as the set of configurations that describe your cluster: a few examples are the number of control plane and worker nodes; the type of Machine hardware that underlies various nodes; regional distribution across nodes or node pools. You may also see this described as "cluster shape" elsewhere.
+
+## Scale out control plane nodes
+
+A common Kubernetes cluster maintenance activity is scaling out (or in) the number of control plane nodes in response to cluster activity. Because Cluster API configuration interfaces are in fact Kubernetes resources, there are a lot of ways to do this. We'll demonstrate using a variety of methods.
+
+### Scale from 1 to 3 control plane nodes
+
+Because we are leveraging the Kubernetes declarative model, we can simply refer to a desired configuration specification and rely upon Cluster API to evaluate what's different between the two.
+
+Assuming that your "`docker-cluster-one`" Cluster still has its original configuration, you should have one control plane node:
+
+```sh
+kubectl --kubeconfig cluster-one.kubeconfig get nodes
+```
+
+Output:
+
+```
+NAME                                             STATUS   ROLES           AGE   VERSION
+docker-cluster-one-md-0-nrh7k-75ddc4778f-vlph9   Ready    <none>          24m   v1.24.6
+docker-cluster-one-xbqb2-tlmpb                   Ready    control-plane   24m   v1.24.6
+```
+
+Let's use the idempotent model to submit a modified configuration of our "`docker-cluster-one`" Cluster with 3 control plane replicas instead of 1. We've provided [a reference yaml of this updated configuration in this repo](yamls/clusters/docker-cluster-one-3-control-plane-replicas.yaml).
+
+If you're on masOS or Linux you can diff this modified spec from the original spec used to create the "`docker-cluster-one`" Cluster:
+
+```sh
+diff yamls/clusters/docker-cluster-one.yaml yamls/clusters/docker-cluster-one-3-control-plane-replicas.yaml
+```
+
+Output:
+
+```
+17c17
+<       replicas: 1
+---
+>       replicas: 3
+```
+
+The above shows that the `yaml` specs are almost identical, with the only change being the `replicas` value on L17. By applying that modified spec to our kind management cluster we can achieve a control plane node scale out operation:
+
+```sh
+kind get kubeconfig --name kubecon-na-22-capi-lab > kubecon-na-22-capi-lab.kubeconfig
+kubectl --kubeconfig kubecon-na-22-capi-lab.kubeconfig apply -f yamls/clusters/docker-cluster-one-3-control-plane-replicas.yaml
```

Comment:
```suggestion
kubectl apply -f yamls/clusters/docker-cluster-one-3-control-plane-replicas.yaml
```

I think we can skip getting a kubeconfig for the mangement cluster 

After we created it, it is the current context. So we can keep it simple for users. kubectl current context is always the mgmt cluster and they then only need --kubeconfig for the workload clusters

Reply:
Ooh boy, you're really testing my OCD here....

Reply:
I'll address this in a follow-up (assuming we wanted to, but otherwise we have to update a few more markdowns to always use the kubeconfig)